x-pack/test/intercepts_api_integration/deployment_agnostic/apis/index.ts

```diff
@@ -0,0 +1,15 @@
+/*
```

Comment:
What's the reason for creating a new `deployment_agnostic` structure under `intercepts_api_integration`? To me, this looks like quite some overhead and the tests could actually be moved to the `x-pack/test/api_integration/deployment_agnostic` directory?

Reply:
There's no specific reason other than wanting to colocate all api tests for intercept api in the same directory, especially that it's possible to do this, the readme for the deployment agnostics test also specified this pattern as the recommend file structure

Reply:
We're currently in the middle of relocating all tests from `x-pack/test` to platform and solution specific directories and adding extra directory levels like this will make things more complicated. And it sounds like there's no specific case for this yet - AFAICS there are only two new test files in this directory now which are deployment agnostic. So these should actually move under the existing `x-pack/test/api_integration/deployment_agnostic` directory. If there are new tests in the future for intercepts which are not deployment agnostic, we can still talk about where to place them - we already made some exceptions and allowed teams to e.g. place their local-only tests there with proper tags in order to keep their tests together.

Reply:
Thanks for the explanation... I made the change and placed them within the `deployment_diagonstic` directory here [`32e0fe2` (#221743)](https://github.com/elastic/kibana/pull/221743/commits/32e0fe2abd024ec0e07af664f16da43da7f292c0), whilst working on that it made me recall why I opted for a separate directory, I couldn't tell if I could just create my own directory or it had to be nested under some other grouping.